File: app/src/main/java/cn/ltt/projectcollection/MainActivity.java
package cn.ltt.projectcollection;

import android.content.Intent;
import android.util.SparseArray;
import android.view.View;

import butterknife.OnClick;
import cn.ltt.projectcollection.base.BaseActivity;
import cn.ltt.projectcollection.beziercurve.BezierCurveActivity;
import cn.ltt.projectcollection.picloading.PictureLoadingActivity;
import cn.ltt.projectcollection.picparticlessplit.PicParticlesSplitActivity;

public class MainActivity extends BaseActivity {

    private SparseArray<Class<? extends BaseActivity>> mActivityArray = new SparseArray<Class<? extends BaseActivity>>() {
        {
            put(R.id.btnPicLoading, PictureLoadingActivity.class);
            put(R.id.btnPicParticlesSplit, PicParticlesSplitActivity.class);
            put(R.id.btnBezierCurve, BezierCurveActivity.class);
        }
    };

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initPage() {

    }

    @OnClick({R.id.btnPicParticlesSplit, R.id.btnPicLoading, R.id.btnBezierCurve})
    public void onClick(View v) {
        startActivity(new Intent(mActivity, mActivityArray.get(v.getId())));
    }
}

File: app/src/main/java/cn/ltt/projectcollection/MainActivity.kt
package cn.ltt.projectcollection

import android.content.Intent
import android.view.View
import cn.ltt.projectcollection.base.BaseActivity
import cn.ltt.projectcollection.beziercurve.BezierCurveActivity
import cn.ltt.projectcollection.dragbubble.DragBubbleActivity
import cn.ltt.projectcollection.picloading.PictureLoadingActivity
import cn.ltt.projectcollection.picparticlessplit.PicParticlesSplitActivity
import kotlinx.android.synthetic.main.activity_main.*

/**
 * ============================================================
 * @author   李桐桐
 * date    2021年04月14日
 * desc    描述
 * ============================================================
 **/
class MainActivity: BaseActivity(), View.OnClickListener {

    private val mActivityArray: HashMap<View,Class<out BaseActivity>> = HashMap()
    private val activity = this

    override fun initPage() {
        with(mActivityArray) {
            put(btnPicLoading.apply { setOnClickListener(activity)}, PictureLoadingActivity::class.java)
            put(btnPicParticlesSplit.apply { setOnClickListener(activity)}, PicParticlesSplitActivity::class.java)
            put(btnBezierCurve.apply { setOnClickListener(activity)}, BezierCurveActivity::class.java)
            put(btnDragBubble.apply { setOnClickListener(activity)}, DragBubbleActivity::class.java)
            put(btnNetRetry.apply { setOnClickListener(activity)}, DragBubbleActivity::class.java)
        }
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun onClick(v: View) {
        startActivity(Intent(mActivity, mActivityArray[v]))
    }

}